static/js/disaster_dashboard/ranking_section.tsx

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for the ranking section of the disaster dashboard
+ */
+
+import _ from "lodash";
+import React from "react";
+
+import {
+  DISASTER_EVENT_INTENSITIES,
+  DISASTER_EVENT_TYPES,
+  DisasterType,
+} from "./constants";
+import { DisasterEventPoint } from "./types";
+
+const RANKING_ITEMS_COUNT = 5;
+
+interface RankingUnitInfo {
+  title: string;
+  prop: string;
+  ranking: DisasterEventPoint[];
+}
+
+interface RankingSectionPropType {
+  disasterEventPoints: DisasterEventPoint[];
+  selectedDisaster: DisasterType;
+  selectedIntensityProp: string;
+  onIntensityPropSelected: (prop: string) => void;
+}
+
+export function RankingSection(props: RankingSectionPropType): JSX.Element {
+  const rankedDisasterCounts = getRankedDisasterCounts(
+    props.disasterEventPoints
+  );
+  const rankingUnits = getRankingUnits(
+    props.disasterEventPoints,
+    props.selectedDisaster
+  );
+  const addSelectionCss =
+    !_.isEmpty(DISASTER_EVENT_INTENSITIES[props.selectedDisaster]) &&
+    DISASTER_EVENT_INTENSITIES[props.selectedDisaster].length > 1;
+  const getRankingUnitClassName = (prop: string) => {
+    if (!addSelectionCss) {
+      return "ranking-unit-no-selection";
+    }
+    return prop === props.selectedIntensityProp
+      ? "ranking-unit-selected"
+      : "ranking-unit";
+  };
+  return (
+    <div className="ranking-section">
+      {props.selectedDisaster === DisasterType.ALL && (
+        <>
+          <h3>Count of Events</h3>
+          {rankedDisasterCounts.map(({ disaster, count }) => {
+            return (
+              <div key={"count-" + disaster} className="ranking-unit-item">
+                {disaster}: {count}
+              </div>
+            );
+          })}
+        </>
+      )}
+      {(!_.isEmpty(rankingUnits) ||
+        props.selectedDisaster !== DisasterType.ALL) && (
+        <>
+          <h3>Rankings</h3>
+          {rankingUnits.map((rankingUnit) => {
+            return (
+              <div
```

Comment:
We should make this regular `<a>` links and work on a url structure for navigating the dashboard for the next iteration.

Reply:
added TODO in https://github.com/datacommonsorg/website/pull/1832